ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,563 @@
+        "get events --sort-by='{.lastTimestamp}'",
+        namespace,
+    )
+
+    scheduled_time = None
+    pod_no = num_of_pods
+    accepted_creation_time = 12
+
+    for line in logs:
+        logger.info(line)
+        if "Scheduled" in line:
+            scheduled_time = int(line.split()[0][:-1])
+        elif "Created" in line:
+            created_time = int(line.split()[0][:-1])
+            creation_time = scheduled_time - created_time
+            logger.info(f"POD number {pod_no} was created in {creation_time} seconds.")
+            if creation_time > accepted_creation_time:
+                raise ex.PerformanceException(
+                    f"POD creation time is {creation_time} and is greater than "
+                    f"{accepted_creation_time} seconds."
+                )
+            pod_no -= 1
+
+
+def measure_pvc_creation_time(interface, pvc_size, pvc_objs, start_time):
+    """
+    Measures and Logs PVC Creation Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        pvc_size (int) : Size of each PVC in GB.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+        start_time (str) : Formatted time from which and on to search the relevant logs.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC creation time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    msg_prefix = f"Interface: {interface}, PVC size: {pvc_size}."
+    accepted_creation_time = 1
+    pvc_no = 0
+    for pvc_obj in pvc_objs:
+        creation_time = performance_lib.measure_pvc_creation_time(
+            interface, pvc_obj.name, start_time
+        )
+
+        logger.info(
+            f"{msg_prefix} PVC number {pvc_no + 1} of interface {interface} was created in {creation_time} seconds."
+        )
+        if creation_time > accepted_creation_time:
+            raise ex.PerformanceException(
+                f"{msg_prefix} PVC creation time is {creation_time} and is greater than "
+                f"{accepted_creation_time} seconds."
+            )
+        pvc_no += 1
+
+
+def measure_pvc_deletion_time(interface, num_of_pvcs, pvc_size, pvc_objs, pv_objs):
+    """
+    Measures and Logs PVC Deletion Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        num_of_pvcs (int) : Number of PVCs we have created.
+        pvc_size (int) : Size of each PVC in GB.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+        pv_objs (list) : List of PV names created for the respective PVCs.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC deletion time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    msg_prefix = f"Interface: {interface}, PVC size: {pvc_size}."
+    accepted_deletion_time = 1
```

Comment:
accepted_deletion_time is set based on the interface, do we still need this line?

Reply:
We don't need this line, but it shows a warning while using this variable later int function as 'Local variable 'accepted_deletion_time' might be referenced before assignment'

Reply:
You will not see the warning if you replace the if else code with below line
accepted_deletion_time = 2 if interface == constants.CEPHFILESYSTEM else 1

Reply:
Ack